src/lib/support/BufferReader.h

```diff
@@ -290,46 +367,80 @@ class Reader
      */
     template <typename T>
     void RawReadLowLevelBeCareful(T * retval);
+};
+
+} // namespace LittleEndian
+
+namespace BigEndian {
 
+/**
+ *  @class Reader
+ *
+ *  Simple reader for reading big-endian things out of buffers.
+ */
+class Reader : public BufferReader
+{
+public:
     /**
-     * Advance the Reader forward by the specified number of octets.
-     *
-     * @param len The number of octets to skip.
+     * Create a buffer reader from a given buffer and length.
      *
-     * @note If the len argument is greater than the number of available octets
-     *       remaining, the Reader will advance to the end of the buffer
-     *       without entering a failed-status state.
+     * @param buffer The octet buffer from which to read.  The caller must ensure
+     *               (most simply by allocating the reader on the stack) that
+     *               the buffer outlives the reader. If `buffer` is nullptr,
+     *               length is automatically overridden to zero, to avoid accesses.
+     * @param buf_len The number of octets in the buffer.
      */
-    Reader & Skip(size_t len)
-    {
-        len = std::min(len, mAvailable);
-        mReadPtr += len;
-        mAvailable = static_cast<size_t>(mAvailable - len);
-        return *this;
-    }
+    Reader(const uint8_t * buffer, size_t buf_len) : BufferReader(buffer, buf_len) {}
 
-private:
     /**
-     * Our buffer start.
+     * Create a buffer reader from a given byte span.
+     *
+     * @param buffer The octet buffer byte span from which to read.  The caller must ensure
+     *               that the buffer outlives the reader.  The buffer's ByteSpan .data() pointer
+     *               is is nullptr, length is automatically overridden to zero, to avoid accesses.
```

Comment:
```suggestion
     *               that the buffer outlives the reader.  If the buffer's ByteSpan .data() pointer
     *               is nullptr, length is automatically overridden to zero, to avoid accesses.
```